server/tests/endpoints/certification/test_certification.py

```diff
@@ -94,7 +96,66 @@ def test_hardware_mismatch(generator: DataGenerator, test_client: TestClient):
                 "firmware_revision": None,
             },
             "os": {
-                "distributor": "Canonical Ltd.",
+                "distributor": "Ubuntu",
+                "version": "24.04",
+                "codename": "noble",
+                "kernel": {"name": None, "version": "5.7.1-generic", "signature": None},
+            },
+            "processor": {
+                "identifier": [],
+                "frequency": 2400,
+                "manufacturer": "AMD",
+                "version": "AMD EPYC 3251 8-Core Processor",
+            },
+        },
+    )
+
+    assert response.status_code == 200
+    assert response.json() == {"status": "Not Seen"}
+    assert "Hardware cannot be found" in caplog.text
+    assert "board model: sample board" in caplog.text
+
+
+def test_bios_mismatch(
+    caplog: LogCaptureFixture,
+    generator: DataGenerator,
+    test_client: TestClient,
+):
+    """
+    Setup where board model and vendor match but bios does not match.
+    The mismatch should be logged
+    """
+    vendor = generator.gen_vendor(name="Known Vendor")
+    board = generator.gen_device(
+        vendor,
+        identifier="dmi:0001",
+        name="FW00Q",
+        version="0001A",
+        category=DeviceCategory.BOARD,
+        bus=BusType.dmi,
+    )
```

Comment:
It would be great if the generator had more specific device creation methods. For instance `gen_board`. This way we could prefill the category and we could also add some defaults that would generate a dummy board for us without needing to repeat it in every test. Doing that, this code would just look like this:
```python
board = generator.gen_device(vendor)
```